lom/src/pluginmgr/pluginmgr_common/pluginmgr_helper_test.go

```diff
@@ -97,348 +97,2504 @@ func (m *MockPlugin) GetPluginID() plugins_common.PluginId {
+            "plugin1": &MockPlugin{},
+            "plugin2": &MockPlugin{},
+        },
+    }
 
-    pluginManager, _ := NewPluginManager(mockClient)
+    // Call the function being tested
+    plugin, ok := pluginManager.getPlugin("plugin1")
 
-    // Execution of run
-    _ = pluginManager.run()
+    // Assert that the plugin was retrieved successfully
+    assert.NotNil(t, plugin)
+    assert.True(t, ok)
 
-    // Verification
-    str := "pluginmgr_common/pluginmgr_helper.go:127:Error RecvServerRequest() : nil"
-    rx := regexp.MustCompile(`^pluginmgr_common/pluginmgr_helper\.go:\d+:Error RecvServerRequest\(\) : nil$`)
-    assert.Regexp(t, rx, str)
-    mockClient.AssertExpectations(t)
+    // Call the function being tested with a plugin name that doesn't exist
+    plugin, ok = pluginManager.getPlugin("plugin3")
+
+    // Assert that the plugin was not retrieved successfully
+    assert.Nil(t, plugin)
+    assert.False(t, ok)
 }
 
-// Test Run() function
-func TestRun_ActionRequestPluginNotFound_LogError(t *testing.T) {
-    // Setup
-    mockClient := new(mockClientTx)
-    mockClient.On("RecvServerRequest").Return(&lomipc.ServerRequestData{
-        ReqType: lomipc.TypeServerRequestAction,
-        ReqData: &lomipc.ActionRequestData{
-            Action: "nonexistent_plugin", // invalid plugin name
+func TestGetPluginMetadata(t *testing.T) {
+    // Create a PluginManager object
+    pluginManager := &PluginManager{
+        pluginMetadata: map[string]plugins_common.IPluginMetadata{
+            "plugin1": &plugins_common.PluginMetadata{},
+            "plugin2": &plugins_common.PluginMetadata{},
         },
-    }, nil)
+    }
 
-    mockClient.On("RegisterClient", mock.Anything).Return(nil) // pass anything as first argument
+    // Call the function being tested with a plugin name that does exist
+    pluginMetadata, ok := pluginManager.getPluginMetadata("plugin1")
+
+    // Assert that the plugin metadata was retrieved successfully
+    assert.NotNil(t, pluginMetadata)
+    assert.True(t, ok)
+
+    // Call the function being tested with a plugin name that doesn't exist
+    pluginMetadata, ok = pluginManager.getPluginMetadata("plugin3")
+
+    // Assert that the plugin metadata was not retrieved successfully
+    assert.Nil(t, pluginMetadata)
+    assert.False(t, ok)
+}
+
+func TestSetShutdownStatus(t *testing.T) {
+    // Create a PluginManager object
+    pluginManager := &PluginManager{}
+
+    // Call the function being tested with a value of true
+    pluginManager.setShutdownStatus(true)
+
+    // Assert that the isActiveShutdown variable was set to true
+    assert.True(t, pluginManager.getShutdownStatus())
+
+    // Call the function being tested with a value of false
+    pluginManager.setShutdownStatus(false)
+
+    // Assert that the isActiveShutdown variable was set to false
+    assert.False(t, pluginManager.getShutdownStatus())
+}
+
+func TestGetShutdownStatus(t *testing.T) {
```

Comment:
Just a suggestion: you can follow naming pattern as Test_<structName>_MethodName() so that its easy to know which struct contains this method.

Reply:
Don't know underscore can be in the name. Corrected the methods with "_". test_functionname